ape_infura/provider.py

```diff
@@ -34,6 +35,27 @@ def __init__(self):
 
 class Infura(Web3Provider, UpstreamProvider):
     network_uris: dict[tuple[str, str], str] = {}
+    api_keys: set[str] = set()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.load_api_keys()
+
+    def load_api_keys(self):
+        self.api_keys = set()
+        for env_var_name in _ENVIRONMENT_VARIABLE_NAMES:
+            if env_var := os.environ.get(env_var_name):
+                self.api_keys.update(set(key.strip() for key in env_var.split(",")))
+
+        if not self.api_keys:
+            raise MissingProjectKeyError()
+
+    def __get_random_api_key(self) -> str:
+        """
+        Get a random api key a private method. As self.api_keys are unhashable so have to typecast into list
```

Comment:
this line is still too long for `flake8`.
Please address all linting issues!

Reply:
you can honestly probably delete the second sentence, it is normal for sets to not be hashable